infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -268,7 +259,27 @@ public void handleTokenQuery(CorfuPayloadMsg<TokenRequest> msg,
     public synchronized void resetServer(CorfuPayloadMsg<Long> msg, ChannelHandlerContext ctx, IServerRouter r,
                                          boolean isMetricsEnabled) {
         long initialToken = msg.getPayload();
-        globalLogTail.set(initialToken);
+
+        //
+        // if the sequencer is reset, then we can't know when was
+        // the latest update to any stream or conflict parameter.
+        // hence, we want to conservatively abort any transaction with snapshot time
+        // preceding the reset-time of this sequencer.
+        //
+        // Therefore, we remember the new start tail.
+        // We empty the cache of conflict parameters.
+        // We set the wildcard to the new start tail.
+        //
+        // Note, this is correct, but conservative (may lead to false abort).
+        // It is necessary because we reset the sequencer.
+        //
+        if (initialToken > globalLogTail.get()) {
```

Comment:
I think now only the synchronized methods be public, the other methods should be made private. This is to prevent any bugs that can get introduced down the line.